### ISSUE TITLE
Incomplete translations in Backend and Frontend - Magento 2.4.3

### ISSUE THREAD

**Issue by vertic4l** [magento/magento2#33772]
### Preconditions (*)
1. Upgraded from Magento 2.4.1 to 2.4.3

### Steps to reproduce (*)
1. set admin user locale to german

### Expected result (*)

1. Backend is fully translated as it was with Magento 2.4.1

### Actual result (*)

1. Backend is partially translated

---

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by vertic4l** [comment]
<img width="1559" alt="Bildschirmfoto 2021-08-12 um 18 47 52" src="https://user-images.githubusercontent.com/1771763/129236294-c5f1f46f-96d7-4757-881b-01a96488fed2.png">

**Comment by vertic4l** [comment]
Current js-translation.json for the backend with Magento 2.4.3

<img width="901" alt="Bildschirmfoto 2021-08-12 um 18 51 05" src="https://user-images.githubusercontent.com/1771763/129236804-dc82ef42-697d-4107-a774-2038123a0615.png">

and here the old js-translation.json from Magento 2.4.1
<img width="1083" alt="Bildschirmfoto 2021-08-12 um 18 52 13" src="https://user-images.githubusercontent.com/1771763/129236881-bd779982-4efd-4d29-adc1-97b9f7d1e7a7.png">

Over 200+ entries vs 17...

**Comment by vertic4l** [comment]
@mrtuvn Same issue exists for the frontend. there are also just a few translations instead of 200+

**Comment by maksymz** [comment]
Hi, @vertic4l !

Don't think this issue is directly related to #31804. I've just checked the translation in clean Magento 2.4.3 without language packs. I have added a translation for one of the phrases from your previous file and it was successfully added to the `js-translation.json` file. After that I added the following phrases:
- `jQuery.mage.__('You have item changes')`
- `jQuery.mage.__("Test phrase to be 'translated");`
Phrases were translated as expected. Untranslated phrases are not added to the file.
The phrase `$.mage.__("We don't recognize this file extension.")` comes from the JS code defined in the `phtml` file, so not sure that translation works for it will work.  

Are you sure the language pack used for the website is compatible with Magento 2.4.3 and it was successfully installed? Try moving the translation to the core file just to see whether the translations work this way or not. Create a file as it does not exist by default - `/vendor/magento/language-de_de/de_DE.csv`

![Selection_032](https://user-images.githubusercontent.com/13600382/129727191-51d8872a-d401-4635-b64e-f751f3f3d82a.png)
